File: server/Models/Movie.cs
namespace MoviePriceComparison.Models;

// This is our domain model for a movie, used in our application
public class Movie
{
    public string Id { get; set; } = string.Empty;
    public string Title { get; set; } = string.Empty;
    public int Year { get; set; }
    public string Type { get; set; } = string.Empty;
    public string Poster { get; set; } = string.Empty;
    public string Provider { get; set; } = string.Empty; // "Cinemaworld" or "Filmworld"
    
    // Additional properties from MovieDetails
    public string? Plot { get; set; }
    public string? Rated { get; set; }
    public string? Released { get; set; }
    public string? Runtime { get; set; }
    public string? Genre { get; set; }
    public string? Director { get; set; }
    public string? Writer { get; set; }
    public string? Actors { get; set; }
    public decimal? Price { get; set; }
    
    // Create a Movie from a MovieItem response
    public static Movie FromMovieItem(MovieItem item, string provider)
    {
        return new Movie
        {
            Id = item.ID,
            Title = item.Title,
            Year = int.TryParse(item.Year, out int year) ? year : 0,
            Type = item.Type,
            Poster = item.Poster,
            Provider = provider
        };
    }
} 

File: client/src/types/Movie.ts
// Movie type as returned by the movie list endpoint
export interface Movie {
  Title: string;
  Year: string;
  ID: string;
  Type: string;
  Poster: string;
}

// Movie details type as returned by the movie details endpoint
export interface MovieDetails extends Movie {
  Rated: string;
  Released: string;
  Runtime: string;
  Genre: string;
  Director: string;
  Writer: string;
  Actors: string;
  Plot: string;
  Language: string;
  Country: string;
  Metascore: string;
  Rating: string;
  Votes: string;
  Price: string;
}

// Movie with price comparison information
export interface MovieComparison {
  id: string;
  title: string;
  year: string;
  poster: string;
  prices: {
    cinemaworld?: string;
    filmworld?: string;
  };
  cheapestProvider?: "cinemaworld" | "filmworld" | null;
}
